src/Prototype/Exe/Form/Signup.hs

```diff
@@ -87,14 +65,14 @@ signupPage Page {..} = Dsl.SingletonCanvas $ do
                     H.div ! A.class_ "o-form-group" $ do
                       H.label
                         ! A.class_ "o-form-group__label"
-                        ! A.for "email"
+                        ! A.for "email-addr"
                         $ "Email address"
                       H.div
                         ! A.class_ "o-form-group__controls"
                         $ H.input
                         ! A.class_ "c-input"
-                        ! A.id "email"
-                        ! A.name "email"
+                        ! A.id "email-addr"
+                        ! A.name "email-addr"
```

Comment:
I think having separate field names in HTML and in Haskell may be a bit more aesthetic, but it adds a good deal of contextual overhead. I'd strongly be in favour of using the same field names and using Generic derivations.